backend/src/main/java/ai/verta/modeldb/versioning/blob/diff/ConflictGenerator.java

```diff
@@ -69,43 +70,147 @@ public static BlobDiff setConflictBlobsInDiff(
+      }
+      switch (locSpecificBlobDiffB.get(0).getContentCase()) {
+        case CODE:
+          diff2.setCode(getCodeConflictBlob(a.getCode(), b.getCode(), c.getCode()).get(0));
+          break;
+        case CONFIG:
+          diff2.setConfig(getConfigConflictBlob(a.getConfig(), b.getConfig(), c.getConfig()));
+          break;
+        case DATASET:
+          diff2.setDataset(
+              getDatasetConflictBlob(a.getDataset(), b.getDataset(), c.getDataset()).get(0));
+          break;
+        case ENVIRONMENT:
+          diff2.setEnvironment(
+              getEnvironmentonflictBlob(a.getEnvironment(), b.getEnvironment(), c.getEnvironment())
+                  .get(0));
+          break;
+        case CONTENT_NOT_SET:
+          break;
+      }
+      return Arrays.asList(diff1.toProto().build(), diff2.toProto().build());
+    }
+    LOGGER.warn("During conflict generation code should not reach here.");
+    throw new ModelDBException("Exception during conflict generation", Code.INTERNAL);
   }
 
-  private static AutogenEnvironmentDiff getEnvironmentonflictBlob(
+  private static List<AutogenEnvironmentDiff> getEnvironmentonflictBlob(
       AutogenEnvironmentDiff a, AutogenEnvironmentDiff b, AutogenEnvironmentBlob c) {
-    return Utils.removeEmpty(
-        new AutogenEnvironmentDiff()
-            .setCommandLine(
-                getCommandLineDiff(
-                    a == null ? null : a.getCommandLine(),
-                    b == null ? null : b.getCommandLine(),
-                    c == null ? Collections.emptyList() : c.getCommandLine()))
-            .setDocker(
-                getDockerDiff(
-                    a == null ? null : a.getDocker(),
-                    b == null ? null : b.getDocker(),
-                    c == null ? null : c.getDocker()))
-            .setEnvironmentVariables(
-                getEnvVarDiff(
-                    a == null ? null : a.getEnvironmentVariables(),
-                    b == null ? null : b.getEnvironmentVariables(),
-                    c == null ? null : c.getEnvironmentVariables()))
-            .setPython(
-                getPythonDiff(
-                    a == null ? null : a.getPython(),
-                    b == null ? null : b.getPython(),
-                    c == null ? null : c.getPython())));
+
+    AutogenPythonEnvironmentDiff pythonDiff =
+        getPythonDiff(
+            a == null ? null : a.getPython(),
+            b == null ? null : b.getPython(),
+            c == null ? null : c.getPython());
+    AutogenDockerEnvironmentDiff dockerDiff =
+        getDockerDiff(
+            a == null ? null : a.getDocker(),
+            b == null ? null : b.getDocker(),
+            c == null ? null : c.getDocker());
+    AutogenCommandLineEnvironmentDiff cliDiff =
+        getCommandLineDiff(
+            a == null ? null : a.getCommandLine(),
+            b == null ? null : b.getCommandLine(),
+            c == null ? Collections.emptyList() : c.getCommandLine());
```

Comment:
Instead of just `c == null ? Collections.emptyList() : c.getCommandLine());` can we add the same line for the `a` & `b`? i am not sure how much it will helpful.

Reply:
I am not sure if that will help a lot, also I would avoid touching it in this PR since it is just moved code with no new logic